Entities/ADoor.h

```diff
@@ -1,452 +1,278 @@
+		/// <param name="targetPos">The absolute position of the target bitmap's upper left corner in the Scene.</param>
+		/// <param name="mode">In which mode to draw in. See the DrawMode enumeration for the modes.</param>
+		/// <param name="onlyPhysical">Whether to not draw any extra 'ghost' items of this ADoor, indicator arrows or hovering HUD text and so on.</param>
+		virtual void Draw(BITMAP *targetBitmap, const Vector &targetPos = Vector(), DrawMode mode = g_DrawColor, bool onlyPhysical = false) const;
+
+		/// <summary>
+		/// Draws this ADoor's current graphical HUD overlay representation to a BITMAP of choice.
+		/// </summary>
+		/// <param name="targetBitmap">A pointer to a BITMAP to draw on.</param>
+		/// <param name="targetPos">The absolute position of the target bitmap's upper left corner in the Scene.</param>
+		/// <param name="whichScreen">Which player's screen this is being drawn to. May affect what HUD elements get drawn etc.</param>
+		/// <param name="playerControlled">Whether or not this MovableObject is currently player controlled (not applicable for ADoor)</param>
+		virtual void DrawHUD(BITMAP *targetBitmap, const Vector &targetPos = Vector(), int whichScreen = 0, bool playerControlled = false);
+#pragma endregion
+
+	protected:
+
+		static Entity::ClassInfo m_sClass; //!< ClassInfo for this class.
+
+		int m_InitialSpriteAnimDuration; //!< This stores the original SpriteAnimDuration value so we can drive the death spin-up animation using LERP.
+
+		std::list<ADSensor> m_Sensors; //!< All the sensors for detecting Actors approaching the door.
+		Timer m_SensorTimer; //!< Times the exit interval.
+		long m_SensorInterval; //!< The delay between each sensing pass in ms.
+
+		Attachable *m_Door; //!< Actual door module that moves. Owned by this.
+
+		DoorState m_DoorState; //!< Current door action state.
+		DoorState m_DoorStateOnStop; //!< The state this door was in when it was stopped.
+
+		bool m_ClosedByDefault; //!< Whether the closed position is the default.
+
+		Vector m_OpenOffset; //!< Open offset from this' position - these effectively replace the door's parent offset.
+		Vector m_ClosedOffset; //!< Closed offset from this' position - these effectively replace the door's parent offset.
+
+		float m_OpenAngle; //!< Open absolute angle of the door attachable, in radians.
+		float m_ClosedAngle; //!< Closed absolute angle of the door attachable, in radians.
+
+		Timer m_DoorMoveTimer; //!< Timer for opening and closing the door.
+		int m_DoorMoveTime; //!< The time it takes to open or close the door in ms.
+
+		bool m_ResumeAfterStop; //!< Whether the door is starting movement after being forced stopped.
+		bool m_ChangedDirectionAfterStop; //!< Whether the door changed directions while moving between states.
+		double m_DoorMoveStopTime; //!< The elapsed time of m_DoorMoveTimer when the door was forced stopped.
+		
+		Timer m_ResetDefaultTimer; //!< Timer for the resetting to the default state.
```

Comment:
I would rename this slightly to `m_ResetToDefaultTimer` and `m_ResetToDefaultDelay`. Makes it a little clearer just from the variable name.